substrate/primitives/runtime/src/lib.rs

```diff
@@ -364,6 +366,13 @@ pub enum MultiSigner {
 	Sr25519(sr25519::Public),
 	/// An SECP256k1/ECDSA identity (actually, the Blake2 hash of the compressed pub key).
 	Ecdsa(ecdsa::Public),
+	/// Same as `Ecdsa` but its account id is derived based off its eth address instead of its
+	/// pubkey.
+	///
+	/// This is important so that the address matches the address to address mapping in
+	/// `pallet_revive`. This means that the same public key controls two accounts. But
+	/// this is already the case due to `pallet_revive`'s address mapping.
+	Eth(ecdsa::Public),
```

Comment:
I won't go down this rabbit hole further in this time-sensitive point, but I think the original intent of some of the types in `sp-runtime`, like this one, but also `UncheckedExtrinsic`, `Block` and `Header`, was to be an example/default, and every runtime should be free to alter them, but overtime this fell apart to the point that I worry half the PJS/PAPI tooling won't work if you touch these too much. I think you actually do alter `UncheckedExtrinsic` in some of the runtimes now, which is a good example. I suppose the same can be done in theory for this too. 

I think the core issue is that these types are left to be _configurable_ in the runtime, but the tooling/metadata is not flexible towards it. 

Anyways, just some thoughts, no action needed.

Reply:
In hindsight we should have just hard coded all types.